BinaryObjectScanner/Protection/SmartE.cs

```diff
@@ -1,28 +1,84 @@
 ﻿using System.Collections.Generic;
 using BinaryObjectScanner.Interfaces;
 using SabreTools.Matching;
+using SabreTools.Matching.Content;
 using SabreTools.Matching.Paths;
 using SabreTools.Serialization.Wrappers;
 
 namespace BinaryObjectScanner.Protection
 {
-    public class SmartE : IExecutableCheck<PortableExecutable>, IPathCheck
+    public class SmartE : IPathCheck, IExecutableCheck<PortableExecutable>
     {
-        /// <inheritdoc/>
         public string? CheckExecutable(string file, PortableExecutable pex, bool includeDebug)
         {
-            // Get the last section strings, if they exist
-            var sections = pex.Model.SectionTable ?? [];
-            var strs = pex.GetSectionStrings(sections.Length - 1);
-            if (strs != null)
-            {
-                if (strs.Exists(s => s.Contains("BITARTS")))
-                    return "SmartE";
-            }
+           // Only works on stub generated from running the program yourself
+           if (pex.InternalName.OptionalEquals("SmarteSECURE"))
+           {
+               return "SmartE";
+           }
+           var sections = pex.Model.SectionTable ?? [];
 
-            return null;
-        }
+           if (sections.Length > 0)
```

Comment:
Do you want to have the "last section" check after the others? It feels like it would steamroll them in the cases that you care about.

Reply:
Discussion decided on discord. Fixed.